api/controllers/game/rematch.js

```diff
@@ -40,14 +40,37 @@ module.exports = async function (req, res) {
       shouldNewGameBeRanked,
       gameService.GameStatus.STARTED,
     );
-    await Promise.all([
+    const [ , , p0, p1] = await Promise.all([
       Game.updateOne({ id: updatedGame.id }).set({ rematchGame: newGame.id }),
       Game.replaceCollection(newGame.id, 'players').members([newP0Id, newP1Id]),
       User.updateOne({ id: newP0Id }).set({ pNum: 0 }),
       User.updateOne({ id: newP1Id }).set({ pNum: 1 }),
     ]);
     updatedGame.rematchGame = newGame.id;
 
+    //Get all exisiting rematchGames
+    let rematchGames = [];
+    const getRematchGames = async (gameId) => {
+      const [gameToAdd] = await Game.find({ rematchGame: gameId });
```

Comment:
You can use `.findOne()` to avoid the need for destructuring here
```suggestion
      const gameToAdd = await Game.findOne({ rematchGame: gameId });
```